src/integration/java/io/pinecone/integration/dataPlane/UpdateFetchAndQueryServerlessTest.java

```diff
@@ -124,48 +119,54 @@ public void updateAllParamsFetchAndQuerySyncTest() throws InterruptedException {
                     true,
                     true);
 
-            ScoredVectorWithUnsignedIndices scoredVectorV1 = queryResponse.getMatches(0);
+            List<ScoredVectorWithUnsignedIndices> queryResults = queryResponse.getMatchesList();
+            ScoredVectorWithUnsignedIndices scoredVectorV1 = null;
+
+            for (ScoredVectorWithUnsignedIndices vector: queryResults) {
+                if (idToUpdate.equals(vector.getId())) {
+                    scoredVectorV1 = vector;
+                }
+            }
+
             // Verify the correct vector id was updated
+            assertNotNull(scoredVectorV1);
             assertEquals(scoredVectorV1.getId(), idToUpdate);
 
             // Verify the updated values
-            List<Float> valuesList = new ArrayList<>(scoredVectorV1.getValuesList());
-            Collections.sort(valuesList);
-            Collections.sort(valuesToUpdate);
+            List<Float> valuesList = scoredVectorV1.getValuesList();
             assertEquals(valuesToUpdate, valuesList);
 
             // Verify the updated metadata
             assertEquals(scoredVectorV1.getMetadata(), metadataToUpdate);
 
             // Verify the initial sparse values set for upsert operation
-            List<Float> expectedSparseValues = new ArrayList<>(scoredVectorV1.getSparseValuesWithUnsignedIndices().getValuesList());
-            List<Float> actualSparseValues = new ArrayList<>(sparseValuesList.get(0));
-            Collections.sort(actualSparseValues);
-            Collections.sort(expectedSparseValues);
-
-            assertEquals(expectedSparseValues, actualSparseValues);
-        });
+            List<Float> expectedSparseValues = sparseValuesList;
+            List<Float> actualSparseValues = scoredVectorV1.getSparseValuesWithUnsignedIndices().getValuesList();
+            // TODO - See if this is an issue, sparseValues seem unsorted after upsert -> update
+            for (Float value : expectedSparseValues) {
+                assertTrue(actualSparseValues.contains(value));
+            }
```

Comment:
This was the only test I saw this issue in (I'm guessing this is why we had those `Collections.sort()` calls all over before?).

Basically it seemed like the sparseValues could be sorted differently after upserting a vector. We would need to troubleshoot why, for now this validates the values.

Reply:
Maybe we should just use `contains` so we're order-agnostic?

Reply:
Is that possible with two arrays, seeing if one contains the other regardless of ordering? I'm basically just checking each value using `contains` atm.